scripts/run_e2e_tests.py

```diff
@@ -697,15 +702,17 @@ def run_tests(args=None):
                                     # test is retried.
                                     pass # pragma: no cover
                                 return 'flake'
+    if failure_seen and p.returncode != 0 and run_count <= MAX_RETRY_COUNT:
+        python_utils.PRINT('\n\n###### RERUN COUNT %d ######\n\n' % run_count)
+        cleanup_portserver(portserver_process)
+        cleanup()
+        run_tests(run_count=run_count + 1, args=args)
```

Comment:
This recursion seems unnecessary. What if instead we did this:

* Keep `run_tests` as a function that runs the tests only once, i.e. eliminate `run_count`
* Have `run_tests` return `pass` `fail` or `flake` to describe the outcome of the test run
* In `main`, run the tests `MAX_RETRY_COUNT` times like we did before. If the return value of `run_tests` is `pass`, break out of the loop

Reply:
The "retry on flake" code that we currently have doesn't actually work so I didn't want to combine the new retry logic along with that. Removal of the previous code is out of scope for this PR, it will be handled by @DubeySandeep in #11263.

I can remove the recursion used here in favour of an iterative approach.

Reply:
Done.